pkgs/tools/package-management/lix/default.nix

```diff
@@ -3,58 +3,81 @@
   aws-sdk-cpp,
   boehmgc,
   callPackage,
+  fetchgit,
   fetchFromGitHub,
   rustPlatform,
   Security,
+  newScope,
 
   storeDir ? "/nix/store",
   stateDir ? "/nix/var",
   confDir ? "/etc",
 }:
 let
-  boehmgc-nix_2_3 = boehmgc.override { enableLargeConfig = true; };
-
-  boehmgc-nix = boehmgc-nix_2_3.overrideAttrs (drv: {
-    patches = (drv.patches or [ ]) ++ [
-      # Part of the GC solution in https://github.com/NixOS/nix/pull/4944
-      ../nix/patches/boehmgc-coroutine-sp-fallback.patch
-    ];
-  });
-
-  aws-sdk-cpp-nix =
-    (aws-sdk-cpp.override {
-      apis = [
-        "s3"
-        "transfer"
-      ];
-      customMemoryManagement = false;
-    }).overrideAttrs
-      {
-        # only a stripped down version is build which takes a lot less resources to build
-        requiredSystemFeatures = [ ];
-      };
+  makeLixScope =
+    {
+      lix,
+      nix-eval-jobs,
+    }:
+    lib.makeScope newScope (
+      self:
+      lib.recurseIntoAttrs {
```

Comment:
Should these package sets be `recurseIntoAttrs`? Should the top-level container be `recurseIntoAttrs` as well?